platform/keyword-registry/src/main/java/org/codice/ddf/keyword/KeywordRegistryImpl.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.keyword;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KeywordRegistryImpl implements KeywordRegistry {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(KeywordRegistryImpl.class);
+
+  private List<String> keywordList = Collections.emptyList();
+
+  private Map<String, String> mappings = Collections.emptyMap();
+
+  @Override
+  public Optional<String> getKeyword(String keyword) {
+    return Optional.ofNullable(mappings.get(keyword));
+  }
+
+  @Override
+  public Map<String, String> getKeywords() {
+    return mappings;
+  }
+
+  public void setKeywordList(List<String> keywordList) {
+    this.mappings = new HashMap<>();
+    this.keywordList = new ArrayList<>();
+
+    LOGGER.info("Setting up keyword registry with values from [{}]", keywordList);
+
+    for (String keyword : keywordList) {
+      String[] parts = keyword.split("=");
+
+      if (parts.length == 2) {
+        LOGGER.info("Adding entry [{}] to keyword registry", keyword);
+
+        this.mappings.put(parts[0], parts[1]);
```

Comment:
❓ Do you want to handle a case for empty Strings? i.e. 'foo=` or `=bar`
How about duplicates? i.e. 'foo=bar, foo=bar`

Reply:
The `parts.length -- 2` should take care of the first one

Reply:
Oh, the first example, `[foo=]`, yes you're right. But not `[=bar]`

Reply:
So I wrote tests for these cases but I'm not going to include them. Hopefully this will explain why...

```
"=bar".split("=") = ["", "bar"]
"foo=".split("=") = []
```

This behavior isn't what I expected and tests for these wouldn't test for intended functionality so they're not beneficial imo. I'm open to other points of views though